src/python/WMCore/WorkQueue/DataLocationMapper.py

```diff
@@ -13,6 +13,7 @@
 
 from WMCore.Services.DBS.DBSReader import DBSReader
 from WMCore.WorkQueue.DataStructs.ACDCBlock import ACDCBlock
+from WMCore.MicroService.Tools.Common import getPileupDocs
```

Comment:
I think this dependency on the MicroService package might become a problem.
Perhaps the best would be to either:
a) make a copy of that function to this WorkQueue package;
b) or copy/move that function to a common location, e.g.: https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/Rucio/RucioUtils.py

If we decide to go with b), we would have to update any other module importing it though.

Reply:
Considering that we plan to include a similar function in future developments leveraging MSPileup, maybe we could consider moving it to a common location. To avoid immediate conflicts with existing implementation, we could copy it where @amaltaro suggested, and then change the dependency on WMCore.MicroService.Tools.Common in the other scripts later on. I don't know if GH has the opportunity to show every implementation of a given package in a specific repo.

Reply:
I think we can see it here:
https://github.com/search?q=repo%3Admwm/WMCore%20getPileupDocs&type=code

Reply:
@amaltaro For your suggestion and other future imports, is Services kind of a common package that can be imported by any other package, while MicroService, WorkQueue, and other packages more independent from each other (deployed as individual services)?

Reply:
Or is Services directory for interfaces to APIs?

Reply:
Yes, WMCore/Services is a more common package among all the services provided by WMCore:
https://github.com/dmwm/WMCore/blob/master/setup_dependencies.py

Different than WMCore/MicroServices, which is really only packaged for WM microservices.

I still don't think this function fits 100% this module: [WMCore/Services/Rucio/RucioUtils.py](https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/Rucio/RucioUtils.py), but right now it's the best option in my opinion.

Reply:
I am thinking that we could make a new module: WMCore/Services/MSPileup/MSPileupUtils.py, or something like this. Would this be a better option?

Reply:
Yes, Services package provide interfaces to external APIs.

Reply:
Sorry to jump in, but I am interested in the discussion since I am working on a similar issue. I support @d-ylee 's idea to create a new module in WMCore/Services, either as part of our current efforts or later on, depending on our deployment needs

Reply:
Yes, that's a better idea. Even though I think this module would have only that single function. As I think it's better to re-use code instead of implement it 3 times, I'd say go for it. Don't forget providing a basic unit test for that, please.

Reply:
@d-ylee do you prefer we create a new module in WMCore/Services now, or later on when we have already working scripts for our current issues?

Reply:
A new module was created (WMCore/Services/MSPileup/MSPileupUtils.py)